herbert_core/utilities/maths/noisify.m

```diff
@@ -67,15 +78,21 @@
     end
 end
 
-function [fac, is_poisson, ymax] = parse_args(y, e, fac, varargin)
+function [fac, is_poisson, ymax, randfun] = parse_args(y, e, fac, varargin)
     USE_LOCAL_MAX = -inf;
     p = inputParser;
     addRequired(p, 'y', @isnumeric);   % y compulsory
     addRequired(p, 'e', @isnumeric);   % e compulsory if any of remaining optional/parameter arguments present
     numeric_or_poisson = @(x) isnumeric(x) || strcmpi(x,'poisson');
     addOptional(p, 'dist_or_factor', fac, numeric_or_poisson);  % fac (numeric) or distribution ('poisson') optional, default to fac=0.1
     addParameter(p,'maximum_value', USE_LOCAL_MAX, @isnumeric);  % ymax, as parameter 'maximum_value'. Default to internally set negative value.
-    parse(p,y,e,varargin{:});
+    check_function_handle = @(x) isa(x,'function_handle');
+    addParameter(p,'random_number_function', @randn, check_function_handle); 
+    try
+        parse(p,y,e,varargin{:});
+    catch ME
+        disp(ME);
+    end
```

Comment:
It's not immediately clear why we're catching and displaying this error but not exiting. `ME` is a struct, did you want `ME.message`?

Reply:
A good catch. Thanks for the info on ME.message. Should probably be removed; I was trying to track down a failure.